### ISSUE TITLE
hipBLAS compiled for CUDA cannot be found in CMake

### ISSUE THREAD

**Issue by ckendrick** [ROCmSoftwarePlatform/hipBLAS#433]
### What is the expected behavior
- In CMake, `find_package(hipblas)` should work.

### What actually happens
- After updating hip and hipblas to the 4.5 release, `hipblas` can no longer be found be CMake on an NVIDIA platform:
```
CMake Error at /opt/cmake-3.21.3-linux-x86_64/share/cmake-3.21/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  By not providing "Findhip.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "hip", but
  CMake did not find one.

  Could not find a package configuration file provided by "hip" with any of
  the following names:

    hipConfig.cmake
    hip-config.cmake

  Add the installation prefix of "hip" to CMAKE_PREFIX_PATH or set "hip_DIR"
  to a directory containing one of the above files.  If "hip" provides a
  separate development package or SDK, be sure it has been installed.
Call Stack (most recent call first):
  /opt/rocm/hipblas/lib/cmake/hipblas/hipblas-config.cmake:90 (find_dependency)
  CMakeLists.txt:239 (find_package)
```
- This used to work fine on all previous releases.
- A current workaround is to [change this line](https://github.com/ROCmSoftwarePlatform/hipBLAS/blob/develop/library/src/CMakeLists.txt#L157) from `DEPENDS PACKAGE hip` to `DEPENDS PACKAGE HIP`. Then after recompiling and installing hipBLAS, it works as expected.

### How to reproduce
- On an NVIDIA machine, install hip 4.5. 
- Download/clone hipBLAS at the 4.5 release. Compile using `./install.sh -i --cuda`
- In a CMake project, using the following will output the above error.
```
find_package(HIP REQUIRED)
find_package(hipblas REQUIRED)
```

- Looking at `/opt/rocm/hipblas/lib/cmake/hipblas/hipblas-config.cmake:90` shows `find_dependency(hip)` whereas `find_dependency(HIP)` works fine (see workaround above).

### Environment
Ubuntu  20.04 with CUDA 11.5, CMake 3.21.3.
HIP 4.5 installed using debian packages. hipBLAS 4.5 compiled with `--cuda`.

**Comment by amcamd** [comment]
Thank you for reporting this error and a special thanks for providing a workaround. We will reply to this issue when the workaround is implemented to fix the hipBLAS build on an NVIDIA machine with hip 4.5.

**Comment by daineAMD** [comment]
Hi @ckendrick,

I'm trying to reproduce the issue that you're seeing, and I was wondering if you could help. What are the exact steps did you took to install HIP 4.5 on your machine? Is ROCm installed as well?

**Comment by daineAMD** [comment]
This should be resolved in #434 which will make it into a future release. Thank you @ckendrick for letting us know about this issue and having a fix ready.
